docs/javascript/browser-apis-and-jquery.md

```diff
@@ -446,3 +446,308 @@ function increaseCatLikes() {
+        <input type="text" placeholder="Enter Wizard Name" id="wizardName" autocomplete="off">
+        <input id="sub" type="submit" value="Submit">
+    </form>
+
+
+    <script>
+
+    function clickHeading() {
+        $("#heading").show();
+    }
+    // when button is clicked it will show the hidden heading text
+    $("#name").click(clickHeading);
+
+
+    </script>
+
+</div>
+
+</html>
+```
+
+The `dblclick()` method allows us to execute a function when element is double clicked.
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+<title>Events!</title>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+
+</head>
+
+<body>
+
+<div>
+    <button id="name">Click</button>
+    <h1 id="heading">Welcome!</h1>
+    <span class="material-icons-outlined" id="wave">
+        waving_hand
+        </span>
+    <form>
+        <input type="text" placeholder="Enter Wizard Name" id="wizardName" autocomplete="off">
+        <input id="sub" type="submit" value="Submit">
+    </form>
+
+
+    <script>
+
+    function clickTwice() {
+        $("#heading").css("color", "gold");
+    }
+    // double click on heading element to change color
+    $("#name").dblclick(clickTwice);
+
+
+    </script>
+
+</div>
+
+</html>
+```
+
+The `hover()` method allows us to execute a function when element is hovered.
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+<title>Events!</title>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+
+</head>
+
+<body>
+
+<div>
+    <button id="name">Click</button>
+    <h1 id="heading">Welcome!</h1>
+    <span class="material-icons-outlined" id="wave">
+        waving_hand
+        </span>
+    <form>
+        <input type="text" placeholder="Enter Wizard Name" id="wizardName" autocomplete="off">
+        <input id="sub" type="submit" value="Submit">
+    </form>
+
+
+    <script>
+
+    function headingHover() {
+        $("#heading").css({
+         "font-size": "60px",
+        });
+    }
+    //change font size of heading text when user hovers
+    $("#heading").hover(headingHover);
+
+
+
+    </script>
+
+</div>
+
+</html>
+```
+
+The `submit()` method allows us to submit form values.
+
+```html
+<!DOCTYPE html>
+<html>
+
+<head>
+<title>Events!</title>
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js"></script>
+
+</head>
+
+<body>
+
+<div>
+    <button id="name">Click</button>
+    <h1 id="heading">Welcome!</h1>
+    <span class="material-icons-outlined" id="wave">
+        waving_hand
+        </span>
+    <form>
+        <input type="text" placeholder="Enter Wizard Name" id="wizardName" autocomplete="off">
+        <input id="sub" type="submit" value="Submit">
+    </form>
+
+
+    <script>
+
+    function submitName(event) {
+        event.preventDefault();
+        var wizardName = $("#wizardName").val();
+        $("#heading").text(`Welcome ${wizardName}!`);
+    }
+    // add the wizardName to our heading element
```

Comment:
Again, let's be a little more explicit for what this line of code does and not what the side-effects are:

```javascript
// When the `<form>` is submitted, run the `submitName()` function.
```